pkg/suggestion/v1alpha3/hyperopt/base_hyperopt_service.py

```diff
@@ -12,8 +12,15 @@
 
 
 class BaseHyperoptService(object):
-    def __init__(self, algorithm_name=TPE_ALGORITHM_NAME, random_state=None, search_space=None):
+    def __init__(self,
+                 algorithm_name=TPE_ALGORITHM_NAME,
+                 algorithm_conf=None,
+                 search_space=None):
         self.algorithm_name = algorithm_name
+        self.algorithm_conf = algorithm_conf or {}
+        # pop common configurations
+        random_state = self.algorithm_conf.pop('random_state', None)
```

Comment:
[nits] I prefer to use `.get('random_state', None)` because `.pop()` has a side-effect. I think it is not expected behavior from the callers.

Reply:
Pop here means to split common ones out of `algorithm_conf`. Then, we can regard the whole `algorithm_conf` dict as **kwargs of suggest method.

Reply:
OK.  I don't have a strong opinion about this, so please feel free to resolve this conversation 👍